### PR TITLE
fix: fix watch bug and event report bug

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
- 修复watch client 协程泄露
- 修复event 上报网关下线情况导致panic问题
- pprof 路由添加认证

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: pkg/apisix/synchronizer/etcd/resource.go [hunk 1]
```diff
@@ -171,94 +172,101 @@ func (e *resourceStore) parseResource(key, value []byte) (resource apisix.Apisix
...
-	ch := e.client.Watch(
-		c,
-		e.prefix,
-		clientv3.WithPrefix(),
-		clientv3.WithPrevKV(),
-		clientv3.WithRev(e.currentRevision),
-	)
-
+	var ch clientv3.WatchChan
+	newWatch := true
```
Comment (zhu327) [inline pkg/apisix/synchronizer/etcd/resource.go:176]: needCreateChan

File: pkg/registry/etcdadapter.go [hunk 2]
```diff
@@ -249,24 +250,33 @@ func (r *EtcdRegistryAdapter) List(ctx context.Context, key ResourceKey, obj cli
...
-		clientv3.WithPrevKV(),
-		clientv3.WithRev(r.currentRevision),
-	)
+	watchCtx, cancel := context.WithCancel(ctx)
+
 	retCh := make(chan *ResourceMetadata)
+
+	var etcdWatchCh clientv3.WatchChan
+
+	newWatchFlag := true
```
Comment (zhu327) [inline pkg/registry/etcdadapter.go:259]: needCreateChan

**Review by zhu327 — APPROVED** [review]

File: pkg/registry/etcdadapter.go (hunk 2 above)
Comment (Han-Ya-Jun) [reply pkg/registry/etcdadapter.go:259]: done

File: pkg/apisix/synchronizer/etcd/resource.go (hunk 1 above)
Comment (Han-Ya-Jun) [reply pkg/apisix/synchronizer/etcd/resource.go:176]: done

File: pkg/registry/etcdadapter.go [hunk 3]
```diff
@@ -416,6 +336,74 @@ func (r *EtcdRegistryAdapter) Watch(ctx context.Context) <-chan *ResourceMetadat
...
+			"value",
+			string(event.Kv.Value),
+		)
+		// trace
+		metadata, err := r.extractResourceMetadata(string(event.Kv.Key))
+		eventCtx, span := trace.StartTrace(metadata.Ctx, "registry.EventPut")
+		defer span.End()
+		if err != nil {
+			span.RecordError(err)
+			return &metadata, err
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply pkg/registry/etcdadapter.go:358]: done

**Pull request merged**